commit 3fc88eaf31b0490a5238627448b5fc18fc2e76ad (refs/changes/11/3511/1)
Author: Bryan Cantrill <bryan@joyent.com>
Date:   2018-02-28T00:51:39+00:00 (1 year, 7 months ago)
    
    OS-6705 thread spinning in dp_pcache_poll() induces system hang

diff --git a/usr/src/uts/common/io/devpoll.c b/usr/src/uts/common/io/devpoll.c
index 040a1f9190..e0d37e171b 100644
--- a/usr/src/uts/common/io/devpoll.c
+++ b/usr/src/uts/common/io/devpoll.c
@@ -25,7 +25,7 @@
 
 /*
  * Copyright (c) 2012 by Delphix. All rights reserved.
- * Copyright 2017 Joyent, Inc.
+ * Copyright 2018 Joyent, Inc.
  */
 
 #include <sys/types.h>
@@ -405,6 +405,7 @@ repoll:
 					 * about the fd reassignment.
 					 */
 					pdp->pd_fp = fp;
+					pdp->pd_gen = gen;
 				}
 			}
 			/*
